R/shinywrappers.R

```diff
@@ -229,6 +229,22 @@ as.tags.shiny.render.function <- function(x, ..., inline = FALSE) {
   useRenderFunction(x, inline = inline)
 }
 
+# Get relevant attributes from a render function object.
+renderFunctionAttributes <- function(x) {
+  attrs <- c("outputFunc", "outputArgs", "hasExecuted", "cacheHint")
```

Comment:
Does it matter that this appears to be missing `srcref`? Would it make sense to do something like:

```r
attrs <- names(attributes(renderText({})))
```

So we can avoid hard-coding the attributes?

Reply:
The purpose of this function is to pull out the attributes from a render function so that we can hoist them into a function that wraps the original render function. We don't want to carry along the `srcref` attribute - that would probably cause problems.

Maybe:

```R
attrs <- setdiff(names(attributes(x)), c("srcref", "wholeSrcref", "srcfile", "class"))
```

Although I do wonder if there are some other weird attributes that we might have missed here. Now that I think about it some more, I think we should leave the code as is, because the specified are the attributes we want to hoist up; we don't use others, so bringing them along could cause problems with no real benefits.